core/amber/src/main/scala/edu/uci/ics/texera/web/resource/ResultDownloadResource.scala

```diff
@@ -148,7 +138,10 @@ object ResultDownloadResource {
 
     // upload the content asynchronously to avoid long waiting on the user side.
     // may change to thread pool
-    new Thread(() => uploadResult(sheetService, sheetId, result)).start()
+    new Thread(() => {
+      uploadHeader(sheetService, sheetId, schema)
+      uploadResult(sheetService, sheetId, result)
+    }).start()
```

Comment:
I see that we send the response immediately without waiting for the actual upload to complete. It's better to send another response when the upload is actually completed. 
Also there is no exception handling here. Ideally we should add a catch all and send an error message to frontend.

But this change might be big and involve changing this to use websocket, since we can't send many http responses. So maybe we can change it if we actually meet any problems later.

Reply:
currently, there is no good way for the frontend to show backend errors. I suggest we have a snack bar to indicate those states with WebSocket.

For now, the error will only be in the log.

Reply:
I view it less a frontend issue, as we can easily use an notification box to show this error message, something like this: https://ng.ant.design/components/notification/en

It's more about chaniging this http api to a websocket api so that the user notified of the progress or any errors

Reply:
it is websocket api now. maybe I can change this in later PR as I need this fix soon.